Add main screen and add customize navigation

VitaliiDiadchenko opened the pull request:
Add the main screen and add customize navigation

VitaliiDiadchenko committed:
update package .idea

VitaliiDiadchenko committed:
Add FragmentMoviesDetails.kt
Add FragmentMovieList.kt
Add interface FragmentListener.kt
Add fragment_movies_list.xml
Fix ageRatingBackground in movieDetailScreen

VitaliiDiadchenko committed:
Remove package .idea

VitaliiDiadchenko committed:
Update README.md

an-yurii commented on app/src/main/java/com/vitaliidiadchenko/mykotkinapplication/FragmentMoviesDetails.kt:
```diff
@@ -0,0 +1,44 @@
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if(context is FragmentListener) {
+            listener = context
+        }
```
Лаконичнее будет
```suggestion
        listener = context as? FragmentListener
```

an-yurii commented on app/src/main/java/com/vitaliidiadchenko/mykotkinapplication/FragmentMoviesList.kt:
```diff
@@ -0,0 +1,41 @@
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        if (context is FragmentListener) {
+            listener = context
+        }
```
Лаконичнее будет
```suggestion
        listener = context as? FragmentListener
```

an-yurii commented on app/src/main/java/com/vitaliidiadchenko/mykotkinapplication/MainActivity.kt:
```diff
@@ -3,15 +3,26 @@ package com.vitaliidiadchenko.mykotkinapplication
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        R.layout.activity_main
```
`setContentView(...)` нужно вернуть

an-yurii commented on app/src/main/res/layout/fragment_movies_list.xml:
```diff
@@ -0,0 +1,160 @@
+                    app:layout_constraintTop_toTopOf="parent"
+                    app:srcCompat="@drawable/img_small_poster" />
+
+                <ImageView
```
Градиент лучше наложить по-другому. Вью привязать к изображению.
```
                <View
                    android:layout_width="0dp"
                    android:layout_height="0dp"
                    android:background="@drawable/gradient_view_card"
                    app:layout_constraintBottom_toBottomOf="@+id/imgTitlePoster"
                    app:layout_constraintEnd_toEndOf="@+id/imgTitlePoster"
                    app:layout_constraintStart_toStartOf="@+id/imgTitlePoster"
                    app:layout_constraintTop_toTopOf="@+id/imgTitlePoster" />
```

Градиент 
```
<?xml version="1.0" encoding="utf-8"?>
<shape xmlns:android="http://schemas.android.com/apk/res/android"
    android:shape="rectangle" >
    <gradient
        android:angle="90"
        android:endColor="@android:color/transparent"
        android:startColor="@color/background" />
    <corners android:radius="0dp" />
</shape>
```

an-yurii reviewed (approved):
Отличная работа)

VitaliiDiadchenko committed:
Change gradient above img card and img detail poster
Fix fun onAttach in FragmentMoviesList.kt and FragmentMoviesDetails.kt
Fix fun onCreate in MainActivity.kt, added setContentView(...)

VitaliiDiadchenko committed:
Merge remote-tracking branch 'origin/Add_main_screen_and_add_customize_navigation' into Add_main_screen_and_add_customize_navigation
